dsc_lib/src/functions/path.rs

```diff
@@ -31,8 +31,15 @@ impl Function for Path {
         debug!("Executing path function with args: {:?}", args);
 
         let mut path = PathBuf::new();
+        let mut first = true;
         for arg in args {
             if let Value::String(s) = arg {
+                // if first argument is a drive letter, add it with a separator suffix as PathBuf.push() doesn't add it
```

Comment:
Given the drive path in middle case, this isn't really a check for the first argument, just the first argument that is a drive letter. If it was checking for the first argument, `first` would need to be set to `false` regardless of the value of the `arg` after the first iteration.

Consider updating this comment or maybe renaming `first` for clarity?

Reply:
So what's happening with `PathBuf.push()` is that if you do: `'c:','windows'`, you'd expect `c:\windows`, but instead, you get `c:windows` on Windows.  So the check here is only if the first element is a drive letter.  However, based on their docs, if you do: `'a','c:','b'`, then you actually get `c:\b` as the driver letter in the middle is interpreted as absolute.  Not how I would have designed it, but this is the current behavior based on their docs.

Reply:
Actually, thinking about this a bit, the alternative is to have `systemRoot()` return a trailing slash and just keep the `pathBuf` behavior and not have any special handling here.  I think I prefer this although it does mean `systemRoot()` doesn't match `%SYSTEMDRIVE%`, but I think it's ok.